wrangler.toml

```diff
@@ -2,6 +2,11 @@ name = "ubiquibot-worker"
 main = "src/worker.ts"
 compatibility_date = "2023-12-06"
 
+[env.dev]
+kv_namespaces = [
+    { binding = "PLUGIN_CHAIN_STATE", id = "a50ef0a143e449c087b6be613a84e5ba" },
```

Comment:
It is my expectation that we will handle all storage from within plugins. What is the kernel seeking to remember? Can you elaborate on your decision here?

Reply:
For one event type there can be multiple plugin chains, so when the plugins return data to the kernel, the kernel needs to know which plugin chain is running and which plugin in the chain is currently executing. 
For example:
```yml
 issues_label.added:

    - name: "Help Menu"
      description: "This will parse the config and display the help menu for slash commands."
      command: "^\/help$"
      example: "/help"
      uses:
        - ubiquibot/plugin-A
        - ubiquibot/plugin-B

    - name: "Assistive Pricing"
      description: "Set pricing based on Time and Priority labels."
      uses:
        - ubiquibot/plugin-B
        - ubiquibot/plugin-C
        - ubiquibot/plugin-A
```

Another reason is because the config can change when the plugin is running so we need to save the plugin chain at the invocation of the first plugin in the chain.

It also has a nice side feature of preventing plugins from sending data to the kernel without any invocation from the kernel